src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -54,14 +54,14 @@ public NamedTypeSymbol CurrentType
             }
         }
 
-        // current method, possibly a lambda, or null if not available
-        private MethodSymbol _currentMethod;
-        public MethodSymbol CurrentMethod
+        // current method, possibly a lambda or local function, or null if not available
+        private MethodSymbol _currentMethodOrNestedFunction;
+        public MethodSymbol CurrentMethodOrNestedFunction
```

Comment:
The new name is an eyeful that is distracting at its use sites. Perhaps `CurrentFunction`?

Reply:
That would work. @agocke Ok with `CurrentFunction` too?

Reply:
Filed https://github.com/dotnet/roslyn/issues/25768
I'll fix shortly if Andy is ok with Neal's proposal.